### ISSUE TITLE
If you don't have enough data, consider whether there is other data that can help pre-train a model.

### ISSUE THREAD

**Issue by cgreene** [Benjamin-Lee/deep-rules#8]
Often we lack a large amount of labeled data, but sometimes there are abundant unlabeled data. We could touch on pre-training approaches (autoencoder pre-training, etc) as well as transfer + refinement (though you have that in #6 so transfer could just be added there). Perhaps this ends up subsumed into a broader point as people contribute their tips.

**Comment by julianstanley** [comment]
I think it would be great to combine this issue with #6 into a more general heading, maybe along the lines of "Don't reinvent the wheel". That can include things like:

1. Pretraining (as is mentioned in this comment), we can site Razavian ([arxiv](https://arxiv.org/abs/1403.6382)) to support that pretraining is useful, but cite something like Yosinski ([nips](http://papers.nips.cc/paper/5347-how-transferable-are-features-in-deep-n%E2%80%A6)) to remind that pretraining is definitely context-dependent.

2. Available tools (Tensorflow and other open source libraries are often expanding. Check to make sure that someone hasn't already built a tool that solves your problem)

3. Finding similar data. There is a lot of already-available biological data from places like GEO. Training (or perhaps more importantly, testing) a model on already-existing data may help increase the ability of the model to generalize to other contexts.

**Comment by evancofer** [comment]
To many, one of the attractive features of DL is that it offers a high degree of flexibility (with respect to input feature representation) while remaining simple to implement and train. As such, DL may sometimes be a useful approach despite limited training data. In those cases, it is probably a good idea to use regularization methods (e.g. dropout, weight decay, input noise, etc).
